index.js

```diff
@@ -207,7 +229,10 @@ function allBoundsOff() {
 
 function allBoundsOn() {
 
-    for (let i = 0; i < markers.length; i++) { 
+    for (let j = 0; j < bounding_box_area_and_marker_index_pairs.length; j++) { 
+
+        let bounding_box_area_and_marker_index = bounding_box_area_and_marker_index_pairs[j];
+        let i = bounding_box_area_and_marker_index[1];
```

Comment:
Wouldn't it be easier to store the marker itself in `boundin_box_area_and_marker_index_pairs` instead of an index into the `markers` array? That way, we wouldn't need all this indirect indexing and had only one array?

It's your call though... Because `markers` also stores marker type I'm not 100% sure it'd be cleaner. Maybe use a dict or even a `Marker` class that contains area, type and marker?

Reply:
The only reason I did it this way is that a lot of things in the code base are setup to deal with the index already. If I just store the marker itself with the marker type that could work, but I think I would need to change something in almost every function. This just seemed like a good way to go about it without changing much existing code.

Reply:
Okay. As I said, your call.